# Evaluation

## 1) Expliquer ce qu'est un KPI

Un KPI (Key Performance Indicator) est un indicateur clé de performance.

Ce sont des variables qui permettent de mesurer l'efficacité par rapport à un objectif défini.

## 2) Importer les librairies numpy et pandas

In [29]:
import numpy as np
import pandas as pd

## 3) Importer le dataset 'eval.csv'

In [30]:
path = r'eval_data.csv'
df = pd.read_csv(path)

## 4) Afficher les 10 premières lignes du dataset

In [31]:
df.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

## 5) Afficher la dimention du dataset

In [32]:
df.shape

(4622, 6)

## 6) Afficher et décrire les colonnes de notre dataset

In [33]:
df.dtypes

Unnamed: 0             int64
order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

## 7) Afficher les statistiques descriptives du dataset

In [34]:
df.describe()

Unnamed: 0     order_id     quantity
count  4622.000000  4622.000000  4622.000000
mean   2310.500000   927.254868     1.075725
std    1334.400802   528.890796     0.410186
min       0.000000     1.000000     1.000000
25%    1155.250000   477.250000     1.000000
50%    2310.500000   926.000000     1.000000
75%    3465.750000  1393.000000     1.000000
max    4621.000000  1834.000000    15.000000

## 8) Quel est l'item le plus commandé ?

In [51]:
df[df.duplicated()].head()

Empty DataFrame
Columns: [Unnamed: 0, order_id, quantity, item_name, choice_description, item_price]
Index: []

In [35]:
df.item_name.value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

L'item le plus commandé est "Chicken Bowl" avec 726 commandes.

## 9) Combien d'items ont été commandés au total ?

In [37]:
df.quantity.sum()

4972

Un total de 4972 items à été commandé.

## 10) Combien de chiffre d'affaire à réalisé la société ?

In [38]:
df.head()

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [52]:
df.item_price = df.item_price.str.replace(',', '.')

In [54]:
df.item_price = df.item_price.str[1:].astype('float')

In [55]:
df.head()

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   

                                  choice_description  item_price  
0                                                NaN         NaN  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN         NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98

In [58]:
(df.quantity * df.item_price).sum()

38998.020000000004

La société a réalisé un CA de $38998.02.

## 11) Quel est le revenu moyen par commande ?

In [59]:
(df.quantity * df.item_price).mean()

8.624064573197701

Le revenu moyen par commande est de $8.62.